### ISSUE TITLE
terraform: CKV_AWS_382 false positive for aws_vpc_security_group_egress_rule with ip_protocol=-1

### ISSUE THREAD

**Issue by ragumix** [bridgecrewio/checkov#7372]
**Describe the issue**

False positive for unrestricted egress when using the standalone Terraform resource `aws_vpc_security_group_egress_rule`.
Relevant check ID: **CKV_AWS_382** (implemented via `SecurityGroupUnrestrictedEgressAll`, which inherits from `AbsSecurityGroupUnrestrictedEgress`).

Root cause logic gaps in `AbsSecurityGroupUnrestrictedEgress.scan_resource_conf` and `contains_violation`:

1. For `aws_vpc_security_group_egress_rule` resource, the code only evaluates `contains_violation()` if `from_port` or `to_port` are [present](https://github.com/bridgecrewio/checkov/blob/b1114135b5f9a33922ec290826c744e048c8d770/checkov/terraform/checks/resource/aws/AbsSecurityGroupUnrestrictedEgress.py#L71). When `ip_protocol = "-1"`, Terraform allows to omit port attributes, so the violation path is skipped and the function returns `PASSED` unconditionally.
2. The [code](https://github.com/bridgecrewio/checkov/blob/b1114135b5f9a33922ec290826c744e048c8d770/checkov/terraform/checks/resource/aws/AbsSecurityGroupUnrestrictedEgress.py#L81) looks for the key `protocol`, but this resource uses `ip_protocol`; thus even if evaluated, `protocol` would be `None`.

**Examples**

A resource like the following, which allows all outbound traffic (all protocols, all ports, any IPv4 destination), is reported as PASSED instead of FAILED:

```HCL
resource "aws_vpc_security_group_egress_rule" "ecs_container_egress" {
  security_group_id = aws_security_group.ecs_container_instance.id
  description       = "Allow all egress traffic"
  ip_protocol       = "-1"
  cidr_ipv4         = "0.0.0.0/0"
}
```

Additional failing scenario that is currently also misclassified (if using IPv6 only):

```HCL
resource "aws_vpc_security_group_egress_rule" "wide_open_ipv6" {
  security_group_id = aws_security_group.example.id
  description       = "Allow all ipv6"
  ip_protocol       = "-1"
  cidr_ipv6         = "::/0"
}
```
Expected: FAILED.  
Actual: PASSED (same logic gap).

**Version:**

 - Checkov Version: `3.2.494`

**Comment by ragumix** [comment]
Still valid

**Comment by jathinsn27** [comment]
Hi @gruebel @yuvalmich @Saarett
I've raised a PR that fixes this: #7513. Tests included, no conflicts with main. Would really appreciate it if you could take a look. Thanks!
